makiflow/trainers/nn_render/render_trainer.py

```diff
@@ -306,17 +406,20 @@ def _perform_testing(self, exp_params, save_path, path_to_weights, FPS=25):
 
         print('render video')
         for i in range(len(all_pred)):
-            answer = np.clip(all_pred[i] + 1, 0.0, 2.0)
-            answer = answer * 128
-            answer[:, :, 0] = without_face[i][:, :, 0] + masks[i] * answer[:, :, 0]
-            answer[:, :, 1] = without_face[i][:, :, 1] + masks[i] * answer[:, :, 1]
-            answer[:, :, 2] = without_face[i][:, :, 2] + masks[i] * answer[:, :, 2]
-            answer = np.concatenate([answer, origin_image[i]], axis=1)
+            if self._restore_function is None:
```

Comment:
There could a `default restore function` that is being assigned to the `self._restore_function` before the training to get rid of this redundant code.